src/main/java/dev/jpcode/kits/Kit.java

```diff
@@ -40,26 +45,50 @@ public Optional<Item> displayItem() {
         return Optional.ofNullable(displayItem);
     }
 
-    public void setDisplayItem(Item item) {
+    public void setDisplayItem(@Nullable Item item) {
         this.displayItem = item;
     }
 
-    private static class StorageKey {
+    public Optional<LinkedList<String>> commands() {
+        return Optional.ofNullable(commands);
+    }
+
+    public boolean addCommand(String command) {
+        if (commands != null && commands.contains(command)) return false;
+        if (commands == null) this.commands = new LinkedList<>();
+        this.commands.add(command);
+        return true;
+    }
+
+    public boolean removeCommand(String command) {
+        if (commands().isEmpty() || !Objects.requireNonNull(commands).contains(command)) return false;
+        if (commands.size() == 1) commands = null;
+        else commands.remove(command);
+        return true;
+    }
```

Comment:
While we do save a list allocation for the "no commands" case by making the list `@Nullable`, I think you'll find that a bunch of this code is simplified not-insignificantly by making it non-nullable (defaulted to empty list). (this simplicity is something I've come to value, less room to miss things, easier to read)

Its very appealing to me, for example, to not have to worry about whether the backing collection is null whenever we want to read from or write to it:

```suggestion
    public List<String> commands() {
        return commands;
    }

    public boolean addCommand(String command) {
        if (commands.contains(command)) return false;
        this.commands.add(command);
        return true;
    }

    public boolean removeCommand(String command) {
        if (commands().isEmpty() || !commands.contains(command)) return false;
        else commands.remove(command);
        return true;
    }
```

Reply:
I suppose this piece of guidance is "if `null` is supposed to behave identically to an empty list, use an empty list"